### DISCUSSION TITLE
batchWrite putRequest's are missing condition expressions implied by params.exists

### DISCUSSION THREAD

**Discussion by dmeehan1968** [sensedeep/dynamodb-onetable#440] [General]
If I make a create() call without batchWrites, there is a condition expression related to the params.exists setting (unless null, in which case no condition expression).

If I make the same call, including the batch option, the generated expression (as shown by log: true, and observed behaviour) does not include the condition expression.  As a result, a create() call when in a batch will overwrite, when the default expectation is to fail.

It's not clear if this is intended functionality.  I can appreciate that it might be necessary in the `batchWrite` handling of `UnprocessedItems` to avoid retrying items that fail on condition checks (and hard to distinguish whether existences conditions from user supplied conditions), but as it stands this leads to unexpected behaviour (updates existing items instead of failure, contrary to the non-batch call).

One might take the view that batchWrite retries should only be done when the failure is due to throttling.

**Comment by mobsense** [comment]
(edited comment)

Sorry, I don't think you can have a condition check test with batchWrite
